### DISCUSSION TITLE
Distributed minio data loss

### DISCUSSION THREAD

**Discussion by yingjianjian** [minio/minio#13004] [Q&A]
I currently have 9 disks. When I hot-swappable the disks, I unplug the 5 disks and plug them back in, and the historical data is lost. Is this normal?

I think the historical data should not be lost after inserting it back into the disk

## Your Environment

* Version used (`minio --version`):
* minio version RELEASE.2021-08-05T22-01-19Z
* Server setup and configuration:
* Operating System and version (`uname -a`):

**Comment by harshavardhana** [comment]
With 9 disks you have only 4 parity if you replace 5 disks of course you will loose data. 

You cannot loose more than the parity.

**Reply by yingjianjian** [reply]
I just hot-swapped it, the data was not formatted and the disk was not changed. Some historical data is lost after reinserting

**Reply by harshavardhana** [reply]
@yingjianjian that doesn't make any sense - can you detail out the steps? how did you confirm that data is lost? 

removing 5 drives when the server is running itself is a disaster in itself, MinIO will automatically think that your objects have lost quorum and it will purge them on the other 4 drives. 

At any given point in time, you can never remove more than parity drives if you remove you can't expect data to be present.

**Reply by yingjianjian** [reply]
Proceed as follows:
1. Tested reading and writing after unplugging a hard disk
2. Continue to unplug the hard drive and repeat the operation until it cannot be written after 5 blocks and then resume recovery
3. Later found that some historical data was lost

**Reply by harshavardhana** [reply]
yes, you cannot do that @yingjianjian - removing more disks than parity can lead to data loss you cannot remove disks on an active running server that is more than the parity.

**Reply by yingjianjian** [reply]
Can't you re-insert the disk? The data on the disk is still there. Why would cause the loss of historical data

**Reply by harshavardhana** [reply]
> Can't you re-insert the disk? The data on the disk is still there. Why would cause the loss of historical data

if you are reading while you are re-inserting you have created a situation where you have a loss of parity, a loss of parity is considered a dangling object so that object is purged.

**Reply by yingjianjian** [reply]
Is there any circumvention? And if only 4 disks are pulled out, will there be this problem?

**Reply by harshavardhana** [reply]
> Is there any circumvention? And if only 4 disks are pulled out, will there be this problem?

There will be no problem if you pull out 4 disks - that's the parity with 9 disks.

**Reply by yingjianjian** [reply]
thanks
